packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,198 @@
+
+        const alignmentCSS = {
+          start: `left: anchor(${anchorName} left);`,
+          end: `right: anchor(${anchorName} right);`,
+          top: `top: anchor(${anchorName} top);`,
+          bottom: `bottom: anchor(${anchorName} bottom);`,
+          centerX: `left: anchor(${anchorName} center); translate: -50% 0;`,
+          centerY: `top: anchor(${anchorName} center); translate: 0 -50%;`,
+        }[alignment ?? centerAlignment];
+
+        this.anchorPositioningStyleElement.textContent = `
+          #${this.anchor} {
+            anchor-name: ${anchorName};
+          }
+          #${this.id} {
+            inset: unset;
+            ${directionCSS}
+            ${alignmentCSS}
+            position: absolute;
+          }
+        `;
+
+        // Dispatch a ready event for the anchor polyfill
+        document.dispatchEvent(new CustomEvent('anchor-polyfill'));
+      }
+
+      this.anchorElement.addEventListener('focus', this.handleFocus);
+      this.anchorElement.addEventListener('blur', this.handleBlur);
+      this.anchorElement.addEventListener('mouseenter', this.handleMouseEnter);
+      this.anchorElement.addEventListener('mouseleave', this.handleMouseLeave);
+    }
+  }
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.anchorElement?.removeEventListener('focus', this.handleFocus);
+    this.anchorElement?.removeEventListener('blur', this.handleBlur);
+    this.anchorElement?.removeEventListener('mouseenter', this.handleMouseEnter);
+    this.anchorElement?.removeEventListener('mouseleave', this.handleMouseLeave);
+  }
+
+  /**
+   * Shows the tooltip
+   * @param delay Number of milliseconds to delay showing the tooltip
+   * @internal
+   */
+  public showTooltip(delay: number = this.defaultDelay): void {
+    setTimeout(() => {
+      this.setAttribute('aria-hidden', 'false');
+      // @ts-expect-error Baseline 2024
+      this.showPopover();
+    }, delay);
+  }
+
+  /**
+   * Hide the tooltip
+   * @param delay Number of milliseconds to delay hiding the tooltip
+   * @internal
+   */
+  public hideTooltip(delay: number = this.defaultDelay): void {
+    setTimeout(() => {
+      // Detect if the tooltip or anchor element is still hovered and enqueue another hide
+      if (this.matches(':hover') || this.anchorElement?.matches(':hover')) {
+        this.hideTooltip(delay);
+        return;
+      }
+
+      this.setAttribute('aria-hidden', 'true');
+      // @ts-expect-error Baseline 2024
+      this.hidePopover();
+    }, delay);
+  }
+
+  /**
+   * Show the tooltip on mouse enter
+   */
+  public handleMouseEnter = () => this.showTooltip(this.delay);
+  /**
+   * Hide the tooltip on mouse leave
+   */
+  public handleMouseLeave = () => this.hideTooltip(this.delay);
+  /**
+   * Show the tooltip on focus
+   */
+  public handleFocus = () => this.showTooltip(0);
+  /**
+   * Hide the tooltip on blur
+   */
+  public handleBlur = () => this.hideTooltip(0);
```

Comment:
Two nits here:
* our event handler functions are more often written as `<event name>Handler` - this allows it to match the name of the event it's handling. `handleMouseEnter` -> `mouseenterHandler`
* When the handlers are intended to be added to an element that isn't `this`, it can be helpful to include a reference to the element, ex. `mouseenterHandler` -> `mouseenterAnchorHandler`